### PR TITLE
feat: Admin can create users

### PR DESCRIPTION
**Pull request by jcbcapps**
# SC-2704

## Proposed changes
A CMS admin can create a user and assign a role prior to that user's first login

---

## Reviewer notes
Make sure that there are no other users on your local machine (at least not Ethel Neal).
Log in to the CMS as an admin:
- Click on `Users` and `Create User`
- In the `User Id` field, add in `ETHEL.NEAL.643097412@testusers.cce.af.mil`
- Select the `Author` role from the dropdown
- Click `Save`

Now you should be able to log in as Ethel Neal for the first time and have all Author permissions

Check out the e2e tests: https://github.com/USSF-ORBIT/ussf-portal/pull/342

### Start the system
```sh
yarn services:up
yarn dev
cd ../ussf-portal-client
yarn dev
```

Login to the cms http://localhost:3001

---

### As the original developer, I have

- [ ] Met the acceptance criteria
- [ ] Created new stories in Storybook if applicable
- [ ] Created/modified automated unit tests in Jest
- [ ] Created/modified automated [E2E tests](https://github.com/USSF-ORBIT/ussf-portal)
- [ ] Followed guidelines for zero-downtime deploys, if applicable
- [ ] Use [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) to check for basic a11y issues

### As a reviewer, I have

Check out our [How to review a pull request](https://github.com/USSF-ORBIT/ussf-portal/blob/main/docs/how-to/review-pull-request.md) document.

---

### CODE REVIEW

File: src/lists/User.ts [hunk 1]
```diff
@@ -33,8 +33,8 @@ const User = list(
       labelField: 'userId',
       searchFields: ['userId'],
       description: 'Keystone users',
-      isHidden: false, // TODO - only show complete UI to admin
-      hideCreate: true,
+      isHidden: !isAdmin,
+      hideCreate: (args) => !isAdmin(args.context),
```
Comment (jcbcapps) [inline src/lists/User.ts:37]: I noticed this ancient todo and thought I'd clean it up a little.

Comment (gidjin) [reply src/lists/User.ts:37]: Should it be `(args) => !isAdmin(args)` on line 36?

File: src/lists/User.ts [hunk 2]
```diff
@@ -94,13 +92,7 @@ const User = list(
...
         isFilterable: true,
         access: {
-          // Access can only be set using SLAM groups
-          update: () => false,
-        },
-        ui: {
-          itemView: {
-            fieldMode: () => 'read',
-          },
+          update: isAdmin,
```
Comment (gidjin) [inline src/lists/User.ts:95]: Do you know why when I log in as a CMS Author I can still see the save button enabled on any user as well as the isEnabled button? I don't think it's a big deal as if you try to change it you get an access denied error but was curious.

Comment (jcbcapps) [reply src/lists/User.ts:95]: I don't know why that is, but I can take a look. I have a couple of things I can try.

**Review by gidjin — APPROVED** [review]
LGTM!

**Pull request merged**